src/interpreter/executors/main.rs

```diff
@@ -251,9 +260,7 @@ where
                 &self.balances,
                 &self.params,
             )?;
-
-            let outputs = self.transaction().outputs().len();
-            (0..outputs).try_for_each(|o| self.update_memory_output(o))?;
+            self.update_transaction_outputs()?;
```

Comment:
[nit] move this to the `finalize_outputs()` method, to reduce the risk of this not being called.

Reply:
We don't need to call `self.update_transaction_outputs()?;` for `deploy` method